bounded_sum_known_n/BoundedSumKnownN.tex

```diff
@@ -0,0 +1,394 @@
+    
+    Moreover, $\texttt{v}, \texttt{w}$ are assumed to be $\din$-close. By the definition of the symmetric distance metric in the proof definitions document linked in section \ref{sec:versioned-docs}, this is equivalent to stating that $d_{Sym}(v, w) = |\MultiSet(v) \Delta \MultiSet(w)| \leq \din$. Further, applying the histogram notation,\footnote{Note that there is a bijection between multisets and histograms, which is why the proof can be carried out with either notion. For further details, please consult the proof definitions document in section \ref{sec:versioned-docs}.} it follows that
+    \[
+        d_{Sym}(v, w) = \lVert h_{v} - h_{w}\rVert_1 = \sum_z |h_v(z) - h_w(z)| \leq \din.
+    \]
+    We want to show that
+    \[
+        d_{Abs}(\function(v), \function(w)) \leq d_{Sym}(v, w) \cdot \dfrac{\texttt{U-L}}{2}.
+    \]
+    This would imply that
+    \begin{equation}\label{eq:abs1}
+        d_{Abs}(\function(v), \function(w)) \leq d_{Sym}(v, w) \cdot \dfrac{\texttt{U-L}}{2} \leq \din \cdot \dfrac{\texttt{U-L}}{2},
+    \end{equation}
+    and by the stability relation this will imply that
+    \begin{equation}\label{eq:abs2}
+        d_{Abs}(\function(v), \function(w)) \leq \dout,
+    \end{equation}
+    as we want to see. 
+\end{proof}
+
+\subsection{First proof: using the path property (adjacent pairs approach)}
+
+To show that $d_{Abs}(\function(v), \function(w)) \leq d_{Sym}(v, w) \cdot \frac{\texttt{U-L}}{2}$, we will use the three lemmas described in the section ``The path property of symmetric distance on sized domains" from Section 5.2 in the document \href{https://www.overleaf.com/project/60d214e390b337703d200982}{``List of definitions used in the proofs"}. With these three lemmas, which are applicable to \texttt{MakeBoundedSumN} because \texttt{input\_domain} is a a sized domain and \texttt{input\_metric} is symmetric distance, it suffices to show the following: For all vectors $x, y \in$ \texttt{input\_domain} such that $d_{Sym}(x, y) = 2$, it follows that 
+\[
+d_{Abs}(\texttt{function}(x), \texttt{function}(y)) \leq \texttt{U} - \texttt{L}.
+\]
+By Lemma 5.6 from \href{https://www.overleaf.com/project/60d214e390b337703d200982}{``List of definitions used in the proofs"}, we know that vectors $x, y$ only differ on one element, given that, by assumption, $d_{Sym}(x, y) = 2$. Wlog, let this different element be the $k$-th element of $x$ and $y$, where $x_k = \alpha$, $y_k = \beta$ with $\alpha \neq \beta$.\footnote{The first element of a vector is indexed by 0.} Then,
```

Comment:
Wlog sentence is not correct.  We can't assume that x and y differ on the same coordinate.  What we know from Lemma 5.6 is that MultiSet(x) and MultiSet(y) differ by changing only one element.  That is, there is a vector z such that z is identical to x except for one coordinate such that MultiSet(z)=MultiSet(y).  You can now do the proof as: |sum_i x_i - sum_i y_i| = |sum_i  x_i - sum_i z_i| = |sum_i (x_i-z_i)| = |x_k-z_k| <= |U-L| = U-L.

Reply:
It is important to **explicitly** point out that because we don't have overflows and have an integer type, the sums are computed exactly, and so MultiSet(z)=MultiSet(y) implies sum_i z_i = sum_i y_i.  This is not true for floating point types, for example.